mapsforge-map-awt/src/main/java/org/mapsforge/map/awt/graphics/AwtPaint.java

```diff
@@ -171,6 +185,45 @@ public int getTextWidth(String text) {
         return fontMetrics.stringWidth(text);
     }
 
+    @Override
+    public int getTextWidth(String text, boolean includePadding) {
+        Graphics2D graphics2d = bufferedImage.createGraphics();
+        FontMetrics fontMetrics = graphics2d.getFontMetrics(this.font);
+        graphics2d.dispose();
+        int retVal = fontMetrics.stringWidth(text);
+
+        if (includePadding && retVal > 0) {
+            retVal += getFontPadding(fontMetrics);
+        }
+
+        return retVal;
+    }
+
+    @Override
+    public int getTextWidth(final String text, final int widthMax, boolean includePadding) {
+        return getTextWidth(text, includePadding);
+    }
+
+    /**
+     * Returns total vertical font padding (top+bottom). The value can also be used for horizontal padding.
+     */
+    @Override
+    public int getFontPadding() {
+        Graphics2D graphics2d = bufferedImage.createGraphics();
+        FontMetrics fontMetrics = graphics2d.getFontMetrics(this.font);
+        graphics2d.dispose();
+
+        return getFontPadding(fontMetrics);
+    }
+
+    /**
+     * Returns total vertical font padding (top+bottom). The value can also be used for horizontal padding.
+     * The value is computed as 12.5% of {@code fontMetrics.getHeight()}.
+     */
+    public static int getFontPadding(FontMetrics fontMetrics) {
+        return (int) (0.125 * fontMetrics.getHeight());
```

Comment:
@Sublimis Desktop also has similar font metrics methods to use instead of fixed padding.
(see `AwtPointTextContainer`)

Reply:
> Desktop also has similar font metrics methods to use instead of fixed padding. (see `AwtPointTextContainer`)

Unfortunately, that is not true, Android supports font metrics that are not available on AWT (as far as we know). For the easiest explanation, see here. Android:

![image](https://github.com/user-attachments/assets/ed28ab85-a495-4150-878e-1e2a65b2fee3)


AWT:

![img1](https://github.com/user-attachments/assets/9c3fe346-3b51-43fd-b711-fa5b67a03bdf)

In AWT there's no TOP and BOTTOM that are used to calculate padding on Android. In any case, the exact amount of padding is not essential at all, as long as it's some well-defined small fraction of the letter size. It's only used to separate labels that are very close together. Frankly, it would probably be easier to use a fraction in Android as well. Also please note that such fraction padding cannot be considered a *fixed* padding, as it depends on letter size. As such, it serves its purpose very well (to separate labels).

Or are you concerned about the value and/or hard-coding the fraction constant, 0.125? This could be factored out to a field, or e.g. `Parameters` config class, or even given as an option to the user (although not a very useful option...).

Reply:
@Sublimis Thanks, I have seen this image online.

Why 12.5%, why not 10%?

Reply:
> Why 12.5%, why not 10%?

This is a power-of-two number so has an exact binary representation, but that aside, it also has a more practical property for us. On standard screens (dot pitch values cca 0.25-0.27mm) and with AWT sample app defaults, the calculated padding with 10% constant is only 1 pixel for many labels, while with the 12.5% it results in a padding of 2 (our padding is defined as top+bottom, so this is appealing because each side gets 1 pixel of padding).

Reply:
@Sublimis Thanks, please add this explanation as comment in the code.
Otherwise in the future it will be difficult to understand what happens.